airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretPersistence.java

```diff
@@ -25,7 +24,7 @@ public interface SecretPersistence extends ReadOnlySecretPersistence {
   static Optional<SecretPersistence> getLongLived(final DSLContext dslContext, final Configs configs) throws IOException {
     switch (configs.getSecretPersistenceType()) {
       case TESTING_CONFIG_DB_TABLE -> {
-        final Database configDatabase = new ConfigsDatabaseInstance(dslContext).getAndInitialize();
+        final Database configDatabase = new Database(dslContext);
```

Comment:
To validate my understanding: this should be a safe change to make because the assumption is that the initialization check will have already been performed before this method is called, correct?

Would it make any sense to try to codify that assumption in the implementation somewhere? E.g. set something like a singleton class variable to `true` after it has been verified that the database is available and migrated. And if that variable is `false` when one of these other classes tries to instantiate a Database object like this line is doing, then an exception is raised because we don't ever expect to be in that state? I guess my main concern is that at some point someone will forget to call the database initialized/migrated check in some new application that is added, and they will try to just construct a Database object and use it, leading to undefined behavior that may be hard to debug. 

If this will all be solved by the Dependency Injection framework then feel free to disregard this comment!

Reply:
> To validate my understanding: this should be a safe change to make because the assumption is that the initialization check will have already been performed before this method is called, correct?
> 
> Would it make any sense to try to codify that assumption in the implementation somewhere? E.g. set something like a singleton class variable to `true` after it has been verified that the database is available and migrated. And if that variable is `false` when one of these other classes tries to instantiate a Database object like this line is doing, then an exception is raised because we don't ever expect to be in that state? I guess my main concern is that at some point someone will forget to call the database initialized/migrated check in some new application that is added, and they will try to just construct a Database object and use it, leading to undefined behavior that may be hard to debug.
> 
> If this will all be solved by the Dependency Injection framework then feel free to disregard this comment!

@lmossman The short answer is:  this will be solved by the application framework and is a temporary implied state.  Once we have the framework in place, we will guarantee that the availability check runs before the application begins serving requests.  As it stands now, we are assuming that we have manually run the check in the main/startup method before this code gets invoked.  The risk is that if someone deletes that code or moves it, then the risk you have called out could happen.  Coupling the `Database` class with the initializer would put us right back where we were.